packages/python/plotly/_plotly_utils/basevalidators.py

```diff
@@ -50,6 +50,83 @@ def to_scalar_or_list(v):
         return v
 
 
+plotlyjsShortTypes = {
+    "int8": "i1",
+    "uint8": "u1",
+    "int16": "i2",
+    "uint16": "u2",
+    "int32": "i4",
+    "uint32": "u4",
+    "float32": "f4",
+    "float64": "f8",
+}
+
+int8min = -128
+int8max = 127
+int16min = -32768
+int16max = 32767
+int32min = -2147483648
+int32max = 2147483647
+
+uint8max = 255
+uint16max = 65535
+uint32max = 4294967295
+
+
+def to_typed_array_spec(v):
+    """
+    Convert numpy array to plotly.js typed array spec
+    If not possible return the original value
+    """
+    v = copy_to_readonly_numpy_array(v)
```

Comment:
maybe we could use a separate variable to be more clear? And call it something a little more informative than `v`?

Reply:
(Maybe `np_arr` or something)

Reply:
Short variable names are not bad and often easier to read.
In this case `v` stands for value and used in the other parts of this code.
So I think we could/should keep it as is.